include/PreferenceList.h

```diff
@@ -56,6 +59,17 @@ class PreferenceList {
     /// get pref list element at given index
     [[nodiscard]] PrefListElement at(SizeType index) const;
 
+    //set the ties at the given rank 
+    void set_ties(RankType rank, PreferenceList p); 
+
+    //get the tied elements at the given rank 
+    [[nodiscard]] std::vector<PrefListElement> get_ties(RankType rank) const; 
+
+    //find if the particular rank is tied 
+    bool isTied(RankType rank) const;
+
+    void printList();
```

Comment:
Why do we need printList function? operator<< is already overloaded for this class.

Reply:
I had written it for debugging, but forgot to remove it. Merged with operator<< now